### PR TITLE
New page: <line-style>

### PR DESCRIPTION
**Pull request by estelle**
new enumerated data value page.
https://github.com/mdn/data/pull/663
https://github.com/mdn/browser-compat-data/pull/19573

### CODE REVIEW

File: files/en-us/web/css/line-style/index.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+- `groove`
+  - : Displays a border with a carved appearance. It is the opposite of `ridge`.
+- `ridge`
+  - : Displays a border with an extruded appearance. It is the opposite of `groove`.
+- `inset`
+  - : Displays a border that makes the element appear embedded. It is the opposite of `outset`. When applied to a table cell border and {{cssxref("border-collapse")}} is set to `collapsed`, this value behaves like `groove`.
+- `outset`
+  - : Displays a border that makes the element appear embossed. It is the opposite of `inset`. When applied to a table cell with {{cssxref("border-collapse")}} set to `collapsed`, this value behaves like `ridge`.
+
+> **Note:** `<outline-style>`, used as the value for {{cssxref("outline")}} and {{cssxref("outline-style")}}, is similar to `<line-style>`, but does not support `hidden` and includes an `auto` value. When `auto` is set, the user-agent defined line-style is used.
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:46]:
```suggestion
- `dashed`
  - : Displays a series of short square-ended dashes or line segments. The exact size and length of the segments are not defined by the specification and are implementation-specific.
- `solid`
  - : Displays a single, straight solid line.
- `double`
  - : Displays two straight lines with some space between them. The length of the lines adds up to the pixel size defined by the line's width.
- `groove`
  - : Displays a border with a carved appearance. This value is the opposite of `ridge`.
- `ridge`
  - : Displays a border with an extruded appearance. This value is the opposite of `groove`.
- `inset`
  - : Displays a border that makes the element appear embedded. This value is the opposite of `outset`. When applied to a table cell border and {{cssxref("border-collapse")}} is set to `collapsed`, this value behaves like `groove`.
- `outset`
  - : Displays a border that makes the element appear embossed. This value is the opposite of `inset`. When applied to a table cell with {{cssxref("border-collapse")}} set to `collapsed`, this value behaves like `ridge`.

> **Note:** When `<outline-style>` is used as the value type for {{cssxref("outline")}} and {{cssxref("outline-style")}} properties, it is similar to `<line-style>`, but does not support `hidden` and includes the `auto` value. When `auto` is set, the user-agent defined `<line-style>` value is used.
```
"When `auto` is set" - does this mean when `auto` is specified for `<outline-style>`?

File: files/en-us/web/css/line-style/index.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+## Examples
+
+The first example demonstrates the appearance of all the `<line-style>` values. The second example demonstrates how some colors may display in unexpected ways.
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:50]:
```suggestion
The first example demonstrates all the `<line-style>` keyword values. The second example demonstrates how some line style colors may display in unexpected ways.
```

File: files/en-us/web/css/line-style/index.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+### Line styles
+
+This example shows all the `<line-style>` values as values of the CSS {{cssxref("border-style")}} and {{cssxref("column-rule-style")}} properties.
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:54]:
```suggestion
### Defining line styles

This example shows all the `<line-style>` values as values for the CSS {{cssxref("border-style")}} and {{cssxref("column-rule-style")}} properties.
```

File: files/en-us/web/css/line-style/index.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+#### HTML
+
+We have {{HTMLElement("div")}}s, each with a class representing the `<line-style>` that is being demonstrated.
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:58]:
```suggestion
This example uses multiple `<div>`elements, each with a class representing the `<line-style>` value that is being demonstrated.
```

File: files/en-us/web/css/line-style/index.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+</div>
+<div class="outset">
+  <p>outset</p>
+  <p>a b c d e f g h i j k l m n o p q r s t u v w x y z</p>
+</div>
+```
+
+#### CSS
+
+We declare a width of `7px` and style of `double` for the border and the column-rule for all the paragraphs. We then override `double` values by providing a different `<line-style>` value the `border-style` and `column-rule-style` properties of each paragraph.
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:112]:
```suggestion
In the CSS for this example, the border and the column-rule for all the `<p>` elements is defined to have a width of `7px` and the style value of `double`. For each paragraph, the `double` values are then overridden by specifying a different `<line-style>` value for the `border-style` and `column-rule-style` properties.
```

File: files/en-us/web/css/line-style/index.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+This example shows all the `<line-style>` values as values of the CSS {{cssxref("border-style")}} and {{cssxref("column-rule-style")}} properties.
+
+#### HTML
+
+We have {{HTMLElement("div")}}s, each with a class representing the `<line-style>` that is being demonstrated.
+
+```html
+<div class="<line-style>">
+  <p><line-style></li>
+  <p>a b c d e f g h i j k l m n o p q r s t u v w x y z</li>
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:63]:
```suggestion
  <p><line-style></p>
  <p>a b c d e f g h i j k l m n o p q r s t u v w x y z</p>
```

File: files/en-us/web/css/line-style/index.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+#### Results
+
+{{EmbedLiveSample("Line_styles", "500", "600")}}
+
+Notice the black border is not always black. The next example demonstrates line-style and color choice.
+
+### Line style colors
+
+With some line-style values, the color may not be what you would expect. To create the required "3D" effect of `groove,` `ridge`, `inset`, and `outset`, when displaying these values in black or white, user agents display use different color calculations than any other color-line combinations.
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:197]:
```suggestion
#### Result

{{EmbedLiveSample("Line_styles", "500", "800")}}

Notice that the black border is not always black.

### Defining line styles and colors

This example demonstrates line-style and color choice. With some `<line-style>` keyword values, the color of the line may not be what you expect. To create the required "3D" effect of `groove,` `ridge`, `inset`, and `outset` styles when displaying these values in black or white, user agents use different color calculations than any other color-line combinations.
```

File: files/en-us/web/css/line-style/index.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+
+#### HTML
+
+Each {{HTMLElement("div")}} has a different `border-color` set as an inline [`style`](/en-US/docs/Web/HTML/Global_attributes/style). For example:
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:201]:
```suggestion
This example uses multiple `<div>`elements, each with a different `border-color` set as an inline [`style`](/en-US/docs/Web/HTML/Global_attributes/style).
```

File: files/en-us/web/css/line-style/index.md [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+
+<div style="border-color: #993333"></div>
+<div style="border-color: #339933"></div>
+<div style="border-color: #333399"></div>
+</section>
+```
+
+#### CSS
+
+The four sides of each list item have a different `<line-style>` value, and each list item has a different {{cssxref("color_value", "&lt;color>")}} value. We use [generated content](/en-US/docs/Web/CSS/content) to display the CSS declared inline.
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:239]: the "list item" is not defined in the HTML

File: files/en-us/web/css/line-style/index.md [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+div::before {
+  content: attr(style);
+}
+```
+
+#### Results
+
+{{EmbedLiveSample("Line_style_colors", "500", "400")}}
+
+Notice that the almost-black `#000001` may be different from actual black, and that the contrast between the dark and light edges is more noticable when using lighter colors.
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:266]:
```suggestion
#### Result

{{EmbedLiveSample("Line_style_colors", "500", "400")}}

Notice that the almost-black color of `#000001` may be different from the actual black, and the contrast between the dark and light edges is more noticeable when using lighter colors.
```

File: files/en-us/web/css/line-style/index.md [hunk 11]
```diff
@@ -0,0 +1,280 @@
...
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [CSS background and borders](/en-US/docs/Web/CSS/CSS_Backgrounds_and_Borders) module
+- [CSS multi-column layout](/en-US/docs/Web/CSS/CSS_Columns) module
+- [CSS Basic User Interface](/en-US/docs/Web/CSS/CSS_Basic_User_Interface) module
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:280]:
```suggestion
- [CSS backgrounds and borders](/en-US/docs/Web/CSS/CSS_Backgrounds_and_Borders) module
- [CSS basic user interface](/en-US/docs/Web/CSS/CSS_Basic_User_Interface) module
- [CSS multi-column layout](/en-US/docs/Web/CSS/CSS_Columns) module
```

**Review by dipikabh — CHANGES_REQUESTED** [review]
Thanks @estelle! I've indicated some language edits

File: files/en-us/web/css/line-style/index.md [hunk 12]
```diff
@@ -0,0 +1,291 @@
...
+
+<div style="border-color: #993333"></div>
+<div style="border-color: #339933"></div>
+<div style="border-color: #333399"></div>
+</section>
+```
+
+#### CSS
+
+The four sides of each list item have a different `<line-style>` value, and each list item has a different {{cssxref("color_value", "&lt;color>")}} value. We use [generated content](/en-US/docs/Web/CSS/content) to display the CSS declared inline.
```
Comment (estelle) [inline files/en-us/web/css/line-style/index.md:250]:
```suggestion
The four sides of each `<div>` have a different `<line-style>` value, and each list item has a different {{cssxref("color_value", "&lt;color>")}} value. We use [generated content](/en-US/docs/Web/CSS/content) to display the CSS declared inline.
```

File: files/en-us/web/css/line-style/index.md [hunk 13]
```diff
@@ -0,0 +1,291 @@
...
+
+Notice that the black border is not always black.
+
+### Defining line styles and colors
+
+This example demonstrates line-style and color choice. With some `<line-style>` keyword values, the color of the line may not be what you expect. To create the required "3D" effect of `groove,` `ridge`, `inset`, and `outset` styles when displaying these values in black or white, user agents use different color calculations than any other color-line combinations.
+
+#### HTML
+
+This example uses multiple `<div>`elements, each with a different `border-color` set as an inline [`style`](/en-US/docs/Web/HTML/Global_attributes/style).
```
Comment (estelle) [inline files/en-us/web/css/line-style/index.md:212]:
```suggestion
This example uses multiple {{HTMLElement( "div" )}} elements, each with a different `border-color` set as an inline [`style`](/en-US/docs/Web/HTML/Global_attributes/style).
```

File: files/en-us/web/css/line-style/index.md [hunk 14]
```diff
@@ -0,0 +1,291 @@
...
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [CSS backgrounds and borders](/en-US/docs/Web/CSS/CSS_Backgrounds_and_Borders) module
+- [CSS basic user interface](/en-US/docs/Web/CSS/CSS_Basic_User_Interface) module
+- [CSS multi-column layout](/en-US/docs/Web/CSS/CSS_Columns) module
```
Comment (estelle) [inline files/en-us/web/css/line-style/index.md:291]:
```suggestion
- [CSS multi-column layout](/en-US/docs/Web/CSS/CSS_Columns) module

```

File: files/en-us/web/css/line-style/index.md [hunk 15]
```diff
@@ -0,0 +1,291 @@
...
+spec-urls: https://drafts.csswg.org/css-backgrounds/#typedef-line-style
+---
+
+{{CSSRef}}
+
+The **`<line-style>`** {{glossary("enumerated")}} value type represents keyword values that define the style of a line, or the lack of a line. The `<line-style>` keyword values are used in the following longhand and shorthand [border](/en-US/docs/Web/CSS/css_backgrounds_and_borders) and [column](/en-US/docs/Web/CSS/CSS_Columns) properties:
+
+- {{cssxref("border")}}, {{cssxref("border-style")}}
+- {{cssxref("border-block")}}, {{cssxref("border-block-style")}}
+- {{cssxref("border-block-end")}}, {{cssxref("border-block-end-style")}
```
Comment (estelle) [inline files/en-us/web/css/line-style/index.md:15]:
```suggestion
- {{cssxref("border-block-end")}}, {{cssxref("border-block-end-style")}}
```

File: files/en-us/web/css/line-style/index.md [hunk 16]
```diff
@@ -0,0 +1,291 @@
...
+}
+.outset p {
+  border-style: outset;
+  column-rule-style: outset;
+}
+```
+
+#### Result
+
+{{EmbedLiveSample("Line_styles", "500", "800")}}
```
Comment (estelle) [inline files/en-us/web/css/line-style/index.md:202]:
```suggestion
{{EmbedLiveSample("Defining_line_styles", "500", "800")}}
```

File: files/en-us/web/css/line-style/index.md [hunk 17]
```diff
@@ -0,0 +1,291 @@
...
+
+The first example demonstrates all the `<line-style>` keyword values. The second example demonstrates how some line style colors may display in unexpected ways.
+
+### Defining line styles
+
+This example shows all the `<line-style>` values as values for the CSS {{cssxref("border-style")}} and {{cssxref("column-rule-style")}} properties.
+
+#### HTML
+
+This example uses multiple {{HTMLElement( "<div>" )}} elements, each with a class representing the `<line-style>` value that is being demonstrated.
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:69]:
```suggestion
This example uses multiple {{HTMLElement( "div" )}} elements, each with a class representing the `<line-style>` value that is being demonstrated.
```

I think this is what is breaking the other macros on the page

**Review by dipikabh — CHANGES_REQUESTED** [review]

File: files/en-us/web/css/line-style/index.md [hunk 18]
```diff
@@ -0,0 +1,291 @@
...
+title: <line-style>
+slug: Web/CSS/line-style
+page-type: css-type
+browser-compat: css.types.line-style
+spec-urls: https://drafts.csswg.org/css-backgrounds/#typedef-line-style
+---
+
+{{CSSRef}}
+
+The **`<line-style>`** {{glossary("enumerated")}} value type represents keyword values that define the style of a line, or the lack of a line. The `<line-style>` keyword values are used in the following longhand and shorthand [border](/en-US/docs/Web/CSS/css_backgrounds_and_borders) and [column](/en-US/docs/Web/CSS/CSS_Columns) properties:
```
Comment (dipikabh) [inline files/en-us/web/css/line-style/index.md:11]:
```suggestion
The **`<line-style>`** {{glossary("enumerated")}} value type represents keyword values that define the style of a line, or the lack of a line. The `<line-style>` keyword values are used in the following longhand and shorthand [border](/en-US/docs/Web/CSS/CSS_Backgrounds_and_Borders) and [column](/en-US/docs/Web/CSS/CSS_Columns) properties:
```

**Review by dipikabh — CHANGES_REQUESTED** [review]

**Comment by dipikabh** [comment]
Not sure why this flaw is still getting flagged (I fixed it in https://github.com/mdn/content/pull/26487/commits/d3efdc3ebec9969afce4b21705ff38abadc24fa5):

`Can't resolve /en-US/docs/Web/CSS/css_backgrounds_and_borders`

**Review by dipikabh — APPROVED** [review]
The fixes look good other than the broken link flaw

UPDATE - the flaw seems to have been fixed now

**Pull request merged**